gadget/ondisk.go

```diff
@@ -84,35 +82,33 @@ type sfdiskPartition struct {
 	Name  string `json:"name"`
 }
 
-func isCreatedDuringInstall(p *sfdiskPartition, fs *lsblkBlockDevice, sfdiskLabel string) bool {
-	switch sfdiskLabel {
-	case "gpt":
-		// the created partitions use specific GPT GUID types and set a
-		// specific bit in partition attributes
-		if !creationSupported(p.Type) {
-			return false
-		}
-		for _, a := range strings.Fields(p.Attrs) {
-			if !strings.HasPrefix(a, "GUID:") {
-				continue
-			}
-			attrs := strings.Split(a[5:], ",")
-			if strutil.ListContains(attrs, createdPartitionAttr) {
+// WasCreatedDuringInstall returns if the OnDiskStructure was created during
+// install by referencing the gadget volume. A structure is only considered to
+// be created during install if it is a role that is created during install and
+// the start offsets match. We specifically don't look at anything on the
+// structure such as filesystem information since this may be incomplete due to
+// a failed installation, or due to the partial layout that is created by some
+// ARM tools (i.e. ptool and fastboot) when flashing images to internal MMC.
+func WasCreatedDuringInstall(lv *LaidOutVolume, s OnDiskStructure) bool {
+	// for a structure to have been created during install, it must be one of
+	// the system-boot, system-data, or system-save roles from the gadget, and
+	// as such the on disk structure must exist in the exact same location as
+	// the role from the gadget, so only return true if the provided structure
+	// has the exact same StartOffset as one of those roles
+	for _, gs := range lv.LaidOutStructure {
+		// TODO: how to handle ubuntu-save here? maybe a higher level function
+		//       should decide whether to delete it or not?
+		switch gs.Role {
+		case SystemSave, SystemData, SystemBoot:
+			// then it was created during install or is to be created during
+			// install, see if the offset matches the provided on disk structure
+			// has
+			if s.StartOffset == gs.StartOffset {
```

Comment:
Is checking only the start offset enough?

Reply:
well this function will be run only if ensureCompatibleLayout passed already, so we know that the other things about the structures must haved matched if they exist, so at this point yes using offset is enough